test/nuts/mdapi.nut.ts

```diff
@@ -58,6 +60,38 @@ describe('mdapi NUTs', () => {
     });
   });
 
+  describe('mdapi:deploy:cancel', () => {
+    it('will cancel an mdapi deploy via the stash.json', () => {
+      execCmd('force:source:convert --outputdir mdapi');
+      // TODO: once mdapi:deploy is migrated switch to execCmd
+      const deploy = JSON.parse(exec('sfdx force:mdapi:deploy -d mdapi -w 0 --json', { silent: true })) as {
+        result: { id: string };
+      };
+      const result = execCmd('force:mdapi:deploy:cancel --json');
+      expect(result.jsonOutput.status).to.equal(0);
+      const json = result.jsonOutput.result;
+      expect(json).to.have.property('canceledBy');
+      expect(json).to.have.property('status');
+      expect(getString(json, 'status')).to.equal('Canceled');
+      expect(getString(json, 'id')).to.equal(deploy.result.id);
+    });
+
+    it('will cancel an mdapi deploy via the specified deploy id', () => {
+      execCmd('force:source:convert --outputdir mdapi');
+      // TODO: once mdapi:deploy is migrated switch to execCmd
+      const deploy = JSON.parse(exec('sfdx force:mdapi:deploy -d mdapi -w 0 --json', { silent: true })) as {
+        result: { id: string };
+      };
+      const result = execCmd(`force:mdapi:deploy:cancel --json --jobid ${deploy.result.id}`);
+      expect(result.jsonOutput.status).to.equal(0);
+      const json = result.jsonOutput.result;
+      expect(json).to.have.property('canceledBy');
+      expect(json).to.have.property('status');
+      expect(getString(json, 'status')).to.equal('Canceled');
```

Comment:
this would be better if the `cancel` command's output type were imported and used `execCmd<x>`